docs/decisions/001-strip-whitespaces.md

```diff
@@ -0,0 +1,228 @@
+| **Proposed**  | 2019-03-01                                                     |
+| **Accepted**  | (the date the proposal was accepted/rejected)                  |
+| **Driver**    | @yhuard                                                        |
+| **Approver**  | @degliwe                                                       |
+| **Consulted** | @emeryro, @degliwe, @planctus, kalinchernev                    |
+| **Informed**  | WAAT, contributors                                             |
+
+## Decision
+
+In order to avoid issues related to whitespace, we should strip them by default and only add them when needed.
+
+## Context
+
+Whitespaces can be a source of problems in HTML when not handled correctly.
+
+When we develop our components in the [main repository](https://github.com/ec-europa/europa-component-library), we use React, which [removes most of the whitespaces](https://reactjs.org/docs/jsx-in-depth.html#string-literals-1.). We don't specifically think about whitespaces and how the can impact the end result, since there are none.
+
+With Twig, we need to be careful about whitespaces. When they appear [between inline block elements](https://css-tricks.com/fighting-the-space-between-inline-block-elements/) or around content, they alter the output.
+
+For example,
+
+```html
+Hello <a href="#">world!</a>
+```
+
+Is not the same as:
+
+```html
+Hello
+<a href="#">
+  world!
+</a>
+```
+
+## Consequences
+
+### Expected output
+
+The output of a Twig template should not contain whitespaces — unless explictly required — in the following cases:
+
+- between adjacent tags
+
+  Expect:
+
+  ```html
+  <span></span><span></span>
+  ```
+
+  Avoid:
+
+  ```html
+  <span></span> <span></span>
+  ```
+
+- between a tag and its children
+
+  Expect:
+
+  ```html
+  <span><span></span><span></span></span>
+  ```
+
+  Avoid:
+
+  ```html
+  <span>
+    <span></span>
+    <span></span>
+  </span>
+  ```
+
+- between a tag and its text content
+
+  Expect:
+
+  ```html
+  <span>Hello world!</span>
+  ```
+
+  Avoid:
+
+  ```html
+  <span>
+    Hello world!
+  </span>
+  ```
+
+  ```html
+  <span> Hello world! </span>
+  ```
+
+### Concrete actions
+
+#### Use the `spaceless` tag
+
+Wrap your template within `{% spaceless %}`:
+
+<!-- prettier-ignore -->
+```html
+{#
+  Parameters:
+  - ...
+#}
+
+{% spaceless %}
+
+{# Pre-processing #}
+{% set ... %}
+
+{# Render #}
+<my-component>
+  ...
+</my-component>
+
+{% endspaceless %}
+```
+
+If you need to add a whitespace somewhere in your template, you can do it with: `{% endspaceless %}{{ ' ' }}{% spaceless %}`.
+
+#### Use the whitespace control modifier on your tags when needed
+
+You can trim leading and or trailing whitespaces with `-` (dash), e.g:
+
+- `{{- my_var }}`: remove left whitespace
+- `{{ my_var -}}`: remove right whitespace
+- `{{- my_var -}}`: remove both whitespaces
+
+It makes sense to use it when you print content between 2 tags:
+
+```html
```

Comment:
```suggestion

Between adjacent tags:

```

Reply:
no sure about this specific change, wouldn't it break the markup here? (removing ```html)

Reply:
I must have selected the wrong line, you're right.